twake/backend/node/src/services/workspaces/web/routes.ts

```diff
@@ -132,6 +138,20 @@ const routes: FastifyPluginCallback = (fastify: FastifyInstance, options, next)
     }
   };
 
+  const validateDomain = async (
+    request: FastifyRequest<{ Params: WorkspaceRequest; Body: WorkspaceInviteDomainBody }>,
+  ) => {
+    const { domain } = request.body;
+
+    if (
+      /^((?!-))(xn--)?[a-z0-9][a-z0-9-_]{0,61}[a-z0-9]{0,1}\.(xn--)?([a-z0-9\-]{1,61}|[a-z0-9-]{1,30}\.[a-z]{2,})$/.test(
+        domain,
+      ) === false
+    ) {
+      throw fastify.httpErrors.badRequest("invalid domain");
+    }
```

Comment:
Here we should test this agains list of public domains like this one: https://github.com/sweiz/list-of-free-mail-providers/blob/master/list-of-free-mail-providers.json

To avoid allowing all @gmail.com to join automatically some workspace